yesod-core/Yesod/Core/Class/Yesod.hs

```diff
@@ -366,6 +366,22 @@ sslOnlySessions = (fmap . fmap) secureSessionCookies
     setSecureBit cookie = cookie { setCookieSecure = True }
     secureSessionCookies = customizeSessionCookies setSecureBit
 
+-- | Helps defend against CSRF attacks by setting the SameSite attribute on
+-- session cookies to "Lax".
+laxSameSiteSessions :: IO (Maybe SessionBackend) -> IO (Maybe SessionBackend)
+laxSameSiteSessions = sameSiteSession sameSiteLax
+
+-- | Helps defend against CSRF attacks by setting the SameSite attribute on
+-- session cookies to "Strict".
```

Comment:
I'm happy for this to be in core. It seems like a pretty small piece of functionality and core is where the rest of the CSRF protection stuff is. I don't think potential changes to the spec are a huge deal for `yesod-core`: unless the spec changes so much that the lax/strict options are dropped completely, all of this code is pretty generic.

Even if there was a big change, personally I don't see a huge difference between a new plugin vs keeping these functions in core, and clearly marking them as unstable APIs that may need to change in accordance with the spec.

Maybe that's being too kind to potential breaking changes—I know Michael and Greg have worked really hard to not make major breaking changes. Anyone have thoughts on this?